### DISCUSSION TITLE
threading in native code

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#276] [Ideas]
Let me just circulate a wild under-the-shower idea I just thought of. Given that calls to OS always cause interrupts,, scheduling and general overhead, we would like these on the start of execution. A dynamic call to native code is one of the things we would like to limit; but on the other hand, we would like to be able to access native functionality when needed, in case of availability and/or prohibitive time to reproduce in rexx.

Also given that our code is just compiled code native to the OS and ISA, would it be possible to thread-in (is that an existing term?) native code into our executable image, at the start of execution? So that an rxvm instruction could hand off execution to the native program (after readying the parameters according to linkage conventions for ISA and ABI?).

This would (1) limit overhead and scheduling after start of execution and (2) when we can package the rxbins with their natives, an enormous gain in efficiency over the lifetime of oft-executed scripts (like production environments have).

**Comment by adesutherland** [comment]
It sounds a bit like a build time linking tool?
